bitcoin/src/bip32.rs

```diff
@@ -111,76 +111,77 @@ pub struct Xpub {
 #[cfg(feature = "serde")]
 internals::serde_string_impl!(Xpub, "a BIP-32 extended public key");
 
-/// A child number for a derived key
+/// A BIP32 types of key derivation
 #[derive(Copy, Clone, PartialEq, Eq, Debug, PartialOrd, Ord, Hash)]
 pub enum ChildNumber {
-    /// A normal child number, indexes 0 to 2147483647 (the first half of all possible children)
-    Normal(u32),
-    /// A hardened child number, indexes 2147483648 to 4294967295 (the second half of all possible children)
-    Hardened(u32),
+    /// Enum wrapper for a normal child number
+    Normal(NormalChildNumber),
+    /// Enum wrapper for a hardened child number
+    Hardened(HardenedChildNumber),
 }
 
-/// NormalChildNumber is a ChildNumber that is not hardened
-/// made only to be used internally in `derive_pub()` and `NormalChildIterator`
+/// A child number for a derived key
+/// Indexes 2147483648 to 4294967295 (the second half of all possible children)
 #[derive(Copy, Clone, PartialEq, Eq, Debug, PartialOrd, Ord, Hash)]
-pub struct NormalChildNumber(pub u32);
+pub struct HardenedChildNumber(u32);
```

Comment:
You might want to consider putting these two in their separate submodules to avoid accidentally missing checks.

Reply:
I cannot access private fields in another module than the struct.

Reply:
Yes, like that. You can also move `Display` out of that module and use a method instead of index. Similarly with from.


Basically `NormalChildNumber` should only have public methods `from_index`, `from_number`, `to_index`, `to_number` and the constants in the module, everything else should be outside.

> We would need to implement something like `new()` and `value()`

Yes, just please use the names I recommended.